### PR TITLE
earthly: disable

### PR DESCRIPTION
**Pull request by griswoldthecat**
-------------

#### Debug data

PR generated by the [Earthly build](https://github.com/earthly/earthly/blob/main/release/Earthfile) (target +release-homebrew)

* `RELEASE_TAG=v0.5.18`

* `GIT_USERNAME=griswoldthecat`

* `NEW_URL=https://github.com/earthly/earthly/archive/v0.5.18.tar.gz`

* `NEW_SHA256=b7679ae11f76536033de59c27c22ea6ca426d3e876fa61aa389d0710ac639c81`

* `TAGS=dfrunmount dfrunsecurity dfsecrets dfssh dfrunnetwork`

* `LDFLAGS=-X main.DefaultBuildkitdImage=earthly/buildkitd:v#{version} -X main.Version=v#{version} -X main.GitSha=0cb24b0a8bbe4253e96990b7947e18f684918298 `

* `LDFLAGS1=-X main.DefaultBuildkitdImage=earthly/buildkitd:v#{version} -X main.Version=v#{version} `

* `LDFLAGS2=-X main.GitSha=0cb24b0a8bbe4253e96990b7947e18f684918298 `

### CODE REVIEW

**Review by nandahkrishna — COMMENTED** [review]
I just noticed, but `earthly`'s license isn't open source (`BUSL-1.1` or the Business Source License isn't an open source license according to its own text, and it isn't present in the DFSG license list).
```
Notice

The Business Source License (this document, or the “License”) is not an Open
Source license. However, the Licensed Work will eventually be made available
under an Open Source License, as stated in this License.
```
We earlier disabled a formula because it was relicensed to `BUSL-1.1` (#76788, #78402).

`earthly` was relicensed at `0.5.0` in #70193. Do we need to disable this formula as well?

**Comment by carlocab** [comment]
Yep, this should definitely be disabled. We should also probably add an audit for the `BUSL-x.y` license.

**Comment by vladaionescu** [comment]
Hi @carlocab & @nandahkrishna - before disabling, can you give us a couple of weeks for us to figure out hosting our own tap?

We want to provide a way for our users to install Earthly on Mac without interruption. We weren't aware that source available licenses cannot be provided this way.

**Comment by nandahkrishna** [comment]
I was just going to CC you @vladaionescu, to recommend moving `earthly` to a personal/organisation Tap. I personally don't mind waiting for a while to merge this, but I'd like to hear what Carlo (and maybe other maintainers) have to say.

FWIW, it should be fairly easy to move the formula to your Tap, and these may help you:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

(You will be able to provide binary packages (Bottles) too.)

**Comment by vladaionescu** [comment]
Thanks - even a week or a few days might be just enough for us to migrate over. We haven't worked with a tap before and we want to test that everything works as expected for our users.

**Comment by carlocab** [comment]
A few weeks is probably too long, and I'd really rather not have this PR sit around waiting. I'm not opposed to setting the `disable!` date to no more than a week from now, but this should be merged ASAP so that users are given the chance to see the deprecation warning.

CC @homebrew/core for thoughts, also @Homebrew/plc since this is a licensing issue.

**Comment by vladaionescu** [comment]
A week is much better than one day @carlocab :-) Thanks!

Question: is there a way to inform users of updated installation instructions through the deprecation message?

**Comment by vladaionescu** [comment]
I'm thinking something along the lines of

> Please visit https://earthly.dev/get-earthly for updated installation instructions.

**Comment by carlocab** [comment]
That's not really something we do, in large part because we don't have the resources to maintain such instructions (there are currently over a hundred disabled formulae in Homebrew/core). `brew info earthly` already points to your homepage (users can also visit this with `brew home earthly`) -- they should be able to find it from there.

**Comment by carlocab** [comment]
In any case -- setting up a new tap should be quite easy with `brew tap-new`. It shouldn't require too much (or any at all, even) testing if you're fine with building bottles with GitHub CI nodes. (This will restrict you a bit wrt the OSs you can build on.) Feel free to ask if you have any questions about this.

**Comment by SMillerDev** [comment]
Yeah, we can disable for next week. I think that's the most leeway we can give since this is not a Homebrew issue and the requirements are pretty clearly documented.

File: Formula/earthly.rb
Comment (SMillerDev) [inline Formula/earthly.rb:22]:
```suggestion
  disable! date: "2021-07-15", because: "has an incompatible license"
```

**Review by carlocab — APPROVED** [review]

**Comment by alexcb** [comment]
We setup a new tap outside of homebrew-core; earthly can now be installed via

    brew install earthly/earthly/earthly

**Pull request merged**
